hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -228,7 +229,15 @@ public long gasRequirement(final Bytes bytes) {
     public PrecompileContractResult computePrecompile(
             final Bytes input, @NonNull final MessageFrame frame) {
         prepareFields(frame);
-        prepareComputation(input, updater::unaliased);
+        try {
+            prepareComputation(input, updater::unaliased);
+        } catch (InvalidTransactionException e) {
+            final var haltReason =
+                    NOT_SUPPORTED.equals(e.getResponseCode())
+                            ? HederaExceptionalHaltReason.NOT_SUPPORTED
+                            : HederaExceptionalHaltReason.ERROR_DECODING_PRECOMPILE_INPUT;
+            return PrecompileContractResult.halt(null, Optional.of(haltReason));
```

Comment:
:warning: The `PrecompileResult` we return here in `prepareComputation` is [**ignored in computeCosted**](https://github.com/hashgraph/hedera-services/blob/6e39a60fd82ac9750cc92c7cf0e7ebf521eb9c1c/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java#L212-L213) if it does not represent a successful execution --- we only extract the `output` field from it and ignore `haltReason`.  It appears this has been present before this PR --- we return a halted `PrecompileResult` in other scenarios as well in this method. Subsequently, no exceptional halt reason info will be present anywhere (not in actions, not in `ContractCallResult` in the record). Can we change the return type of `computeCosted` to `Pair<Long, PrecompileResult>` and let upstream components handle it? It appears we handle exceptional halts in the precompile in multiple different ways - [here we directly set the exceptional halt on the frame](https://github.com/hashgraph/hedera-services/blob/6e39a60fd82ac9750cc92c7cf0e7ebf521eb9c1c/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/HTSPrecompiledContract.java#L229-L230), it will be nice to handle it uniformly in all scenarios.

We also do not export a child record for the precompile call, I'm not sure whether we want this or not --- when we ultimately change the code here to set the exceptional halt reason properly in the frame, it will be set for the _child frame associated with the precompile call_. However, since no child record will be exported for this call, the exceptional halt reason will only be present in the contract actions --- the error message for the initial frame will be empty. This does not sound OK to me, what do others think?

I guess exporting a child record is debatable, but we must definitely set the exceptional halt reason properly.

Reply:
Thanks @dimitar-dinev. @stoyan-lime @IvanKavaldzhiev @tinker-michaelj  Thoughts ?

Reply:
Adding child records is a big change to add to 0.35, we need to think this through and make changes in 0.36. 
This PR is targeted to just fix the issue when we don't charge gas when `allowances.isEnabled`  the flag is disabled.

Reply:
Makes sense!